### ISSUE TITLE
Beta in endpoints flexible sample pom.xml causes an error

### ISSUE THREAD

**Issue by frankyn** [GoogleCloudPlatform/java-docs-samples#397]
Issue:

The beta configuration in https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/flexible/endpoints/pom.xml for "gcloud beta app deploy" causes an error.

Possible-Solution:

A fast solution was to remove the <gcloud_app_prefix></gcloud_app_prefix> to it in the pom file.

**Comment by bradfriedman** [comment]
Did you run "mvn gcloud:deploy" from the root of the sample app? If that caused an error, could you provide the contents of the error? Thanks!

**Comment by lesv** [comment]
It appears that the latest `gcloud beta app deploy` command  isn't working well, but the `gcloud app deploy` command now understands the new endpoints directives in `app.yaml`

**Comment by frankyn** [comment]
@bradfriedman my error log: https://gist.github.com/frankyn/ce0933c0fab94cd93d7627549934b294

**Comment by wmwong** [comment]
In the logs there is a note about the project ID that will be used for deployment:
Note that the project ID and version specified in application configuration files (e.g. app.yaml or appengine-web.xml) are ignored. The project ID is set to the Cloud SDK default project and the version is set to the time of deployment.

And then there's this error afterwards:
[INFO] ERROR: (gcloud.beta.app.deploy) Projectsetting [spec-test-ruby-samples] not found: Method not found.

Is spec-test-ruby-samples the project that you are trying to deploy to? It looks like the deploy command can't find this project ID.

**Comment by frankyn** [comment]
That's correct, I was trying to deploy to "spec-test-ruby-samples" and it does exist.

**Comment by wmwong** [comment]
@bradfriedman Any ideas what might be going on with the gcloud beta command?

**Comment by frankyn** [comment]
tl;dr; use 'gcloud auth default-credentials login' with my gmail account, and not a service account. 

The functionality depends on certain permissions that doesn't allow the use of service accounts. I had to change my gcloud settings to use my gmail account using 'gcloud auth default-credentials login'. After following the CLI prompt, the endpoints sample was able to deploy without modifying the pom.xml.
